### PR TITLE
ref(chartOptionSelector): Use new CompactSelect

### PR DESCRIPTION
**Pull request by vuluongj20**
Use `CompactSelect` in `charts/optionSelector`, as  `CompactSelect` is more in sign with our new dropdown menu design and more accessible.

**Before:**
Screen Shot 2022-04-26 at 11 54 43 AM

**After:**
Screen Shot 2022-04-26 at 11 52 47 AM

### CODE REVIEW

File: static/app/components/charts/optionSelector.tsx [hunk 1]
```diff
@@ -1,157 +1,73 @@
...
+type SingleProps = BaseProps & {
+  onChange: (value: string) => void;
+  selected: string;
+  multiple?: false;
+};
+type MultipleProps = BaseProps & {
+  multiple: true;
+  onChange: (value: string[]) => void;
+  selected: string[];
+};
```
Comment (vuluongj20) [inline static/app/components/charts/optionSelector.tsx:25]: consolidating `optionSelector` and `optionCheckboxSelector` into one component, only difference is the presence of the `multiple` prop and the accepted values for `selected` and `onChange`

File: static/app/components/forms/compactSelect.tsx [hunk 2]
```diff
@@ -178,41 +178,51 @@ function CompactSelect<OptionType extends GeneralSelectValue = GeneralSelectValu
...
+    // Wait until the trigger element finishes rendering, otherwise
+    // ResizeObserver might throw an infinite loop error.
+    await new Promise(resolve => window.setTimeout(resolve));
+
+    const newTriggerWidth = triggerRef.current?.offsetWidth;
+    newTriggerWidth && setTriggerWidth(newTriggerWidth);
+  }, [triggerRef]);
+  useResizeObserver({ref: triggerRef, onResize: updateTriggerWidth});
+  // If ResizeObserver is not available, manually update the width
+  // when any of [trigger, triggerLabel, triggerProps] changes.
```
Comment (vuluongj20) [inline static/app/components/forms/compactSelect.tsx:219]: same code as in https://github.com/getsentry/sentry/blob/vl/chart-option-selector/static/app/components/dropdownMenuControlV2.tsx#L125-L145

File: static/app/components/forms/selectOption.tsx [hunk 3]
```diff
@@ -12,47 +13,52 @@ function SelectOption(props: Props) {
...
-      <InnerWrap isFocused={isFocused} isDisabled={isDisabled}>
-        <Indent isMulti={isMulti} centerCheckWrap={verticallyCenterCheckWrap}>
-          <CheckWrap isMulti={isMulti} isSelected={isSelected}>
-            {isSelected && (
-              <IconCheckmark
-                size={isMulti ? 'xs' : 'sm'}
-                color={isMulti ? 'white' : undefined}
-              />
+    <selectComponents.Option className="select-option" {...props}>
+      <Tooltip containerDisplayMode="block" title={tooltip} delay={tooltipDelay ?? 500}>
```
Comment (vuluongj20) [inline static/app/components/forms/selectOption.tsx:28]: adding an optional tooltip, for cases like this

Screen Shot 2022-04-26 at 12 17 56 PM

**Comment by evanpurkhiser** [comment]
Let's split out the changes in how the CompactSelect works from the changes in chart option selector? Two PRs will be easier to review

**Comment by vuluongj20** [comment]
> Let's split out the changes in how the CompactSelect works from the changes in chart option selector? Two PRs will be easier to review

Will do!

**Review by dashed — APPROVED** [review]

**Pull request merged**